edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/publisher/IoTHubMetricsUpload.cs

```diff
@@ -30,8 +30,18 @@ public async Task PublishAsync(IEnumerable<Metric> metrics, CancellationToken ca
             if (data.Length > 0)
             {
                 Message message = this.BuildMessage(data);
-                await this.edgeAgentConnection.SendEventAsync(message);
+
+                try
+                {
+                    await this.edgeAgentConnection.SendEventAsync(message);
+                }
+                catch (Exception ex) when (ex.HasTimeoutException())
```

Comment:
>catch (Exception ex) when (ex.HasTimeoutException()) [](start = 16, length = 52)

This exception will be pretty much swallowed. Should we at least log it?

Reply:
The timeout exception is swallowed, b/c a timeout exception means that there was no internet connection and we should retry. In this case, the worker will log that it is starting to retry and the backoff timeout. It's an unfortunate case of control flow exceptions stemming from our sdk dependency.

Reply:
All other exceptions will propagate back to the worker and be logged without retry.

Reply:
emm, I agree, we should log it

Reply:
added debug log